platform/test.rs

```diff
@@ -174,30 +173,77 @@ fn big_data_with_sender_transfer() {
     thread.join().unwrap();
 }
 
-#[test]
-// This test only applies to platforms that need fragmentation.
+macro_rules! create_with_n_fds {
+    ($name:ident, $n:expr, $size:expr) => (
+        #[test]
+        fn $name() {
```

Comment:
Does this really need to be a macro? Can it just be a helper function that the various tests delegate to?

"Macros are for when you have run out of language" —@littlecalculist

But I think there is still language left :)